### ISSUE TITLE
Validate does not count cleanly-passing products, incorrectly triggering no_products_found error

### ISSUE THREAD

**Issue by jordanpadams** [NASA-PDS-Incubator/validate#1557]
## Checked for duplicates
Yes - I've already checked

## 🐛 Describe the bug
When validating a PDS4 label that passes with zero errors and zero warnings, validate incorrectly reports `error.execution.no_products_found` and shows 0 products in the summary — even though the product appears as PASS in the validation results.

Root cause: In `Report.java:231`, `numPassedProds` is only incremented when `problems.size() - ignoreFromTotalCounts > 0`. When a product validates cleanly, all problems are either absent or classified as `ProblemCategory.GENERAL`/`EXECUTION` (which increment `ignoreFromTotalCounts`), causing the net count to be `<= 0` and the product to never be counted. `getTotalProducts()` then returns 0, triggering the false `no_products_found` error in `ValidateLauncher.java:1566`.

## 🕵️ Expected behavior
A product that validates successfully with no errors or warnings should be counted in the product total and should NOT trigger the `no_products_found` error.

## 📜 To Reproduce
1. Run validate against a valid PDS4 label with user-supplied schema/schematron and skip flags:
```
validate --target /path/to/valid_label.xml \
  --schema /path/to/PDS4_PDS_XYYY.xsd \
  --schematron /path/to/PDS4_PDS_XYYY.sch \
  --skip-context-validation \
  --skip-content-validation
```
2. Observe the product shows `PASS` in Product Level Validation Results
3. Observe the summary reports `0 product(s)` and fires `error.execution.no_products_found`

## 🖥 Environment Info
- Version of this software: v4.1.0-SNAPSHOT
- Operating System: macOS

## 📚 Version of Software Used
v4.1.0-SNAPSHOT

## 🩺 Test Data / Additional context
Related to recent changes in product counting logic introduced in:
- #1548 (PR: #1556)
- #1458 (PR: #1538)

Proposed fix — change `Report.java:231` from:
```java
if (!Utility.isDir(sourceUri.toString()) && problems.size() - ignoreFromTotalCounts > 0) {
```
to:
```java
if (!Utility.isDir(sourceUri.toString())) {
```

## 🦄 Related requirements
- #1548
- #1458

**Comment by jordanpadams** [comment]
Skipping I&T: impacts many tests logging so automated by relation to those.
